### PR TITLE
Adds checks for quickstart CRD

### PR DESCRIPTION
**Pull request by vbnrh**
This commit adds conditions for checking if Quickstart CRD is available. This is done as ConsoleQuickStart is not applied in OCP 4.6 and this will reduce unneccessary error messages in the logs.

Signed-off-by: Vineet Badrinath <vineetbnath@gmail.com>

### CODE REVIEW

File: controllers/storagecluster/quickstarts.go [hunk 1]
```diff
@@ -6,13 +6,24 @@ import (
...
 )
 
 type ocsQuickStarts struct{}
 
 func (obj *ocsQuickStarts) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) error {
+	qscrd := extv1.CustomResourceDefinition{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "consolequickstarts.console.openshift.io", Namespace: ""}, &qscrd)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("No custom resource definition found for consolequickstart. Skipping quickstart initialiazation")
```
Comment (umangachapagain) [inline controllers/storagecluster/quickstarts.go:21]: Probably make this a DEBUG level log. We don't want to see this on every reconcile.

File: controllers/storagecluster/quickstarts.go [hunk 2]
```diff
@@ -6,13 +6,24 @@ import (
...
+			return nil
+		}
+		r.Log.Error(err, "Failed to fetch consolequickstart CRD")
+		return nil
```
Comment (umangachapagain) [inline controllers/storagecluster/quickstarts.go:25]: If it failed to fetch, shouldn't we try to fetch it again and try to deploy?

**Review by umangachapagain — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/quickstarts.go (hunk 1 above)
Comment (jarrpa) [reply controllers/storagecluster/quickstarts.go:21]: Agreed.

File: controllers/storagecluster/quickstarts.go (hunk 2 above)
Comment (jarrpa) [reply controllers/storagecluster/quickstarts.go:25]: If we have a non-IsNotFound error in a client Get() call, this is probably indicative of a larger problem. I would imagine we would want to fail and start over as we do elsewhere.

**Review by jarrpa — CHANGES_REQUESTED** [review]
Overall the change makes sense. Please do move the `go.sum` changes to their own commit.

File: controllers/storagecluster/quickstarts.go (hunk 2 above)
Comment (vbnrh) [reply controllers/storagecluster/quickstarts.go]: returning an err here. it should fail the reconcile and retry again

File: controllers/storagecluster/quickstarts.go (hunk 1 above)
Comment (vbnrh) [reply controllers/storagecluster/quickstarts.go]: changed

**Review by umangachapagain — APPROVED** [review]

File: controllers/storagecluster/quickstarts.go [hunk 3]
```diff
@@ -6,13 +6,23 @@ import (
...
 )
 
 type ocsQuickStarts struct{}
 
 func (obj *ocsQuickStarts) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) error {
+	qscrd := extv1.CustomResourceDefinition{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "consolequickstarts.console.openshift.io", Namespace: ""}, &qscrd)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.V(2).Info("No custom resource definition found for consolequickstart. Skipping quickstart initialiazation")
```
Comment (nbalacha) [inline controllers/storagecluster/quickstarts.go:21]: Nit: typo in "initialiazation"

Comment (vbnrh) [reply controllers/storagecluster/quickstarts.go]: fixed

**Review by jarrpa — APPROVED** [review]
Commits look good... but I'm a little puzzled by the test failures. If they continue today I'd ask that they be investigated. It doesn't make sense that these changes should affect functionality, but you never know.

/lgtm

**Comment by vbnrh** [comment]
just rebased the PR to see if the test passes @jarrpa will need your approval again

File: controllers/storagecluster/storagecluster_controller_test.go [hunk 4]
```diff
@@ -6,19 +6,19 @@ import (
...
 	"github.com/blang/semver"
+	snapapi "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
 	"github.com/noobaa/noobaa-operator/v2/pkg/apis/noobaa/v1alpha1"
 	configv1 "github.com/openshift/api/config/v1"
 	consolev1 "github.com/openshift/api/console/v1"
 	openshiftv1 "github.com/openshift/api/template/v1"
 	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
 	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
-
-	snapapi "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
```
Comment (jarrpa) [inline controllers/storagecluster/storagecluster_controller_test.go:16]: This change is invalid, please revert.

**Review by jarrpa — CHANGES_REQUESTED** [review]

**Review by jarrpa — APPROVED** [review]
/lgtm

File: controllers/storagecluster/storagecluster_controller_test.go (hunk 4 above)
Comment (vbnrh) [reply controllers/storagecluster/storagecluster_controller_test.go]: done

**Pull request merged**